### PR TITLE
Refactoring of start.py script. Closes #1899 and #1866 

### LINKED ISSUE #1899: Remove python script.py to reduce dependencies
We should reduce the overall dependencies to install.
We should transform this script in a bash script for better compatibility.

We should take this refactor to solve [this issue](https://github.com/intelowlproject/IntelOwl/issues/1866) altogether

### PR DESCRIPTION
**Pull request by fgibertoni**
# Description
A new shell script `start` has been added [to replace](https://github.com/intelowlproject/IntelOwl/issues/1899) `start.py`.
It features all the options of the previous script while reducing dependencies (removed `pre-requirements.txt`) and including the `initialize.sh` first execution. `script.py` is still available for retro compatibility.

During this refactoring the support for [Docker Compose V1](https://github.com/intelowlproject/IntelOwl/issues/1866) has been dropped and docker requirements were updated. The support has been removed from `start.py` and `initialize.sh` script.

Doc has been updated accordingly for new usage.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [X] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [X] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ ] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

File: docs/source/Installation.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Installation
 
 ## Requirements
-The project leverages `docker-compose` with a custom python script so you need to have the following packages installed in your machine:
-* [docker](https://docs.docker.com/get-docker/) - v1.13.0+
-* [docker-compose](https://docs.docker.com/compose/install/) - v1.23.2+
+The project leverages `docker compose` with a custom Bash script and you need to have the following packages installed in your machine:
+* [docker](https://docs.docker.com/get-docker/) - v19.03.0+
+* [docker-compose](https://docs.docker.com/compose/install/) - v2.3.4+
 * [python](https://www.python.org/) - v3.6+
```
Comment (mlodic) [inline docs/source/Installation.md:7]: shouldn't we able to remove this python dependency too? If I remember correctly, we needed it only to run the python start.py script

File: start [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+if ! [ -d venv/ ]; then
+  ./initialize.sh
+  if ! [ $? ]; then
+    echo "Failed to install dependencies." >&2
+    exit 1
+  fi
+fi
+source venv/bin/activate
+
```
Comment (mlodic) [inline start:103]: is this venv still necessary?

File: start [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+for value in "${compose_files[@]}" ; do
+  if [ -n "${value}" ]; then
+    to_run+=" -f $value"
+  fi
+done
+if grep "docker" <<< "$(groups)" > /dev/null 2>&1; then
+  docker compose --project-directory docker ${to_run[@]} -p "$project_name" "$cmd_argument" "$@"
+else
+  sudo docker compose --project-directory docker ${to_run[@]} -p "$project_name" "$cmd_argument" "$@"
+fi
```
Comment (mlodic) [inline start:331]: cool

**Review by mlodic — COMMENTED** [review]
great work!

File: start (hunk 2 above)
Comment (fgibertoni) [reply start]: Removed venv activation as no more necessary,

File: docs/source/Installation.md (hunk 1 above)
Comment (mlodic) [reply docs/source/Installation.md]: > Python may still be required in order to run Django manage.py commands though.

But Django commands should be executed inside the `uwsgi` container where python is already installed there. So I think there won't be any real utility

File: docs/source/Installation.md [hunk 4]
```diff
@@ -285,6 +294,12 @@ Maintainers strive to keep the upgrade between major version easy but it's not a
 Below you can find the additional process required to upgrade from each major versions.
 </div>
 
+#### Updating to >=6.0.0 from a 5.x.x version
+IntelOwl v6 introduced some major changes regarding how the project is started.
+Before upgrading, some important things should be checked by the administrator:
+* Docker Compose V1 support has been dropped project-wide. If you are still using a Compose version prior to v2.3.4, please [upgrade](https://docs.docker.com/compose/migrate/) to a newer version or install Docker Compose V2.
+* IntelOwl is now started with the new `start` script that has the same options as `start.py` but is more manageable and has decreased the overall project dependencies. The `start.py` script has now been removed.
```
Comment (mlodic) [inline docs/source/Installation.md:301]:
```suggestion
* IntelOwl is now started with the new bash `start` script that has the same options as the old python `start.py` script but is more manageable and has decreased the overall project dependencies. The `start.py` script has now been removed.
```

**Review by mlodic — APPROVED** [review]
great work!

**Review by mlodic — APPROVED** [review]
we are ready to merge once the last comment is resolved

**Comment by fgibertoni** [comment]
You mean codecov ?

File: docker/scripts/cron/application_restart [hunk 5]
```diff
@@ -1 +1 @@
-0 2 * * 0  root python3 start.py test --all_analyzers down && python3 start.py test --all_analyzers up -d
\ No newline at end of file
+0 2 * * 0  root ./start test down --all_analyzers && ./start test up --all_analyzers -- -d
```
Comment (mlodic) [inline docker/scripts/cron/application_restart:1]: I meant this comment that I thought to have submitted:

I think that this cron has never worked properly. Can we assume an absolute working directory (something like `/opt/IntelOwl`) and launch the start.py from there and then add this consideration in the related documentation? Ty

Comment (fgibertoni) [reply docker/scripts/cron/application_restart]: I should have applied your suggestion. Also updated doc.

**Review by mlodic — APPROVED** [review]

**Pull request merged**